src/math/decibels.c

```diff
@@ -8,6 +8,8 @@
 #include <sof/math/decibels.h>
 #include <stdint.h>
 
+#if CONFIG_MATH_DECIBELS
```

Comment:
For files those are totally "on/off", it's simpler to skip build in CMakeList.txt. I'll do that for trig, decibels, IIR, FIR. Changing that...